superset/connectors/sqla/models.py

```diff
@@ -1165,7 +1169,7 @@ def get_sqla_query(  # pylint: disable=too-many-arguments,too-many-locals,too-ma
                     # if groupby field equals a selected column
                     elif selected in columns_by_name:
                         outer = columns_by_name[selected].get_sqla_col()
-                    else:
+                    elif allow_adhoc_subquery(selected):
```

Comment:
Same here, we need a fallback value for `outer` when `allow_adhoc_subquery(selected)` is false.